src/clj/bluegenes_tool_store/package.clj

```diff
@@ -0,0 +1,33 @@
+(ns bluegenes-tool-store.package
+  (:require [bluegenes-tool-store.tools :as tools]
+            [cheshire.core :as cheshire]
+            [clj-http.client :as client]
+            [taoensso.timbre :refer [infof]]))
+
+(defn get-all-npm-tools
+  "Send an HTTP request to the NPMJS API to get a list of bluegenes tools."
+  []
+  (-> (client/get (str "https://api.npms.io/v2/search"
+                       "?q=scope:intermine"
+                       "+keywords:bluegenes-intermine-tool"))
+      :body
+      (cheshire/parse-string true)
+      :results))
+
+(defn install-all-npm-tools
+  "Call NPM through the shell to install all bluegenes tools from the NPM registry."
+  []
+  (let [packages (mapv (comp :name :package) (get-all-npm-tools))]
+    (tools/sync-sh-req (into ["npm" "install" "--save"] packages)
+                       (tools/tools-path :tools))))
+
+(defn setup-package-json
+  "Checks for package.json containing tools and initialises one if it doesn't exist."
+  []
+  (let [config-path (tools/tools-path :config :file true)]
+    (when-not (.exists config-path)
+      (infof "Tool config %s does not exist. It will automatically be created and all bluegenes tools will be installed." config-path)
```

Comment:
I couldn't quite work out from reading the code what the default path was going to be if there wasn't a tools path variable set. Is it dynamic, or could we add it in a comment?

Reply:
The bluegenes tool path? The default is specified in *config/defaults/config.edn*.

Reply:
a comment above this info output would be a good place to specify the default and how to change it...